### DISCUSSION TITLE
Query bool default bug.

### DISCUSSION THREAD

**Discussion by asyba** [fastapi/fastapi#10690] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.route("/data")
def data(request: Request, clean: bool = False):
    #clean = str_to_bool(clean)
    logger.info(f"Method: {request.method}")
    logger.info(f"URL: {request.url}")
    logger.info(f"Headers: {dict(request.headers)}")
    logger.info(f"Query params: {dict(request.query_params)}")
    logger.info(f"clean: {clean}")
```

### Description

The value of clean is always False, not matter if I called it yes/no/False/false/True/true.
it always setting the default value in this case False.
checking request.query_params I can se the real value and is different from the clean.
 

### Operating System

Linux

### FastAPI Version

0.104.1

### Pydantic Version

2.5.1

### Python Version

Python 3.9.18

**Comment by YuriiMotov** [comment]
Didn't manage to reproduce this. Tried with both (FastAPI 0.104.1, 2.5.1) and current versions (FastAPI 0.115.12, Pydantic 2.11.5).
Everything works well.

The only thing I changed was `@app.get("/data")` instead of `@app.route("/data")`
